packages/sandcastle/buildGallery.js

```diff
@@ -0,0 +1,111 @@
+import { readFileSync, existsSync, writeFileSync } from "fs";
+import { parse } from "yaml";
+import { globbySync } from "globby";
+import { basename, dirname, join } from "path";
+import { exit } from "process";
+
+// TODO: probably need to find a way to integrate this into the vite process to make sure it's
+// built during build
+// May also want to move this out of the sandcastle project so we don't have to duplicate
+// files for the local build, need to think about deploying too...
+
+const galleryDirectory = "./public/gallery";
+
+const yamlFiles = globbySync([
+  `${galleryDirectory}/*/sandcastle.yaml`,
+  `!${galleryDirectory}/list.json`,
+]);
+
+console.log("files", yamlFiles);
+
+/**
+ * @typedef GalleryListItem
+ * @property {string} id
+ * @property {string} title
+ * @property {string} thumbnail
+ * @property {string} description
+ * @property {boolean} isNew
+ */
+
+/**
+ * @type {{entries: GalleryListItem[], legacyIdMap: Object<string, string>}}
+ */
+const output = {
+  entries: [],
+  legacyIdMap: {},
+};
+
+let writeFile = true;
+const check = (condition, messageIfTrue) => {
+  if (condition) {
+    console.error(messageIfTrue);
+    writeFile = false;
+  }
+  return condition;
+};
+
+for (const filePath of yamlFiles) {
+  const file = readFileSync(filePath, { encoding: "utf-8" });
+  const metadata = parse(file);
+  console.log("path", filePath);
+  console.log(metadata);
+  if (!metadata) {
+    console.error(filePath, "is empty!");
+    continue;
+  }
+
+  const { id, title, description, thumbnail } = metadata;
+
+  // Validate metadata
+
+  const slug = basename(dirname(filePath));
```

Comment:
Maybe use [`slugify`](https://www.npmjs.com/package/slugify) to ensure we're covering our bases, and using that as the `id` directly. That way, there's less room for user error.

Also using the slugified version of the path would also be preferable to make sure we don't get some of the ugly URLs we do now, like `3D%20Tiles%20Clipping%20Planes.html`.

Reply:
@ggetz I like the idea of using `slugify` as part of the conversion script from the existing gallery to the new one. However I don't think we need to use it here. I'd rather the files be as static as possible which means they should match the slug format in the first place. If it doesn't the developer should fix it and this build script will be noisy about it. 

The FE needs to know where to access the files and images. The id is used in that route `[base page]/gallery/[id]`. If we were to generate the id as a `slugify` of the directory name it may not match.